app/database/server/models/team_membership.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Q, Relation} from '@nozbe/watermelondb';
+import Model, {Associations} from '@nozbe/watermelondb/Model';
+import {immutableRelation} from '@nozbe/watermelondb/decorators';
+
+import {MM_TABLES} from '@constants/database';
+import User from '@typings/database/user';
+import Team from '@typings/database/team';
+
+const {TEAM, TEAM_MEMBERSHIP, USER} = MM_TABLES.SERVER;
+
+/**
+ * The TeamMembership model represents the 'association table' where many teams have users and many users are in
+ * teams (relationship type N:N)
+ */
+export default class TeamMembership extends Model {
+    /** table (entity name) : TeamMembership */
+    static table = TEAM_MEMBERSHIP;
+
+    /** associations : Describes every relationship to this entity. */
+    static associations: Associations = {
+
+        /** TEAM and TEAM_MEMBERSHIP share a 1:N relationship;  USER can be part of multiple teams */
+        [TEAM]: {type: 'belongs_to', key: 'team_id'},
+
+        /** USER and TEAM_MEMBERSHIP share a 1:N relationship;  A TEAM can regroup multiple users */
+        [USER]: {type: 'belongs_to', key: 'user_id'},
+    };
+
```

Comment:
Add fields for user_id and team_id and create lazy queries to retrieve all users for that team and vice-versa.